internal/pkg/cli/task_run.go

```diff
@@ -134,6 +163,117 @@ func (o *runTaskOpts) Ask() error {
 	return nil
 }
 
+// Execute create or update task resources and run the task.
+func (o *runTaskOpts) Execute() error {
+	if o.image == "" && o.dockerfilePath == "" {
+		o.dockerfilePath = "./Dockerfile"
+	}
+
+	if o.imageTag == "" {
+		o.imageTag = "latest"
+	}
+
+	if err := o.getNetworkConfig(); err != nil {
+		return err
+	}
+
+	if err := o.deployTaskResource(); err != nil {
+		return err
+	}
+
+	//if image is not provided, then we build the image and push to ECR repo
+	if o.image == "" {
+		uri, err := o.pushToECRRepo()
+		if err != nil {
+			return err
+		}
+		o.image = fmt.Sprintf(fmtImageURL, uri, o.imageTag)
+
+		// update image to stack
+		if err := o.deployTaskResource(); err != nil {
+			return err
+		}
+	}
+
+	// TODO: kick off task
+	return nil
+}
+
+func (o *runTaskOpts) getNetworkConfig() error {
+	if o.env != config.EnvNameNone {
+		subnets, err := o.ec2.GetSubnetIDsFromAppEnv(o.AppName(), o.env)
+		if err != nil {
+			return fmt.Errorf("get subnet ids: %w", err)
+		}
+
+		securityGroups, err := o.ec2.GetSecurityGroupsFromAppEnv(o.AppName(), o.env)
+		if err != nil {
+			return fmt.Errorf("get security groups: %w", err)
+		}
+
+		o.subnets = subnets
+		o.securityGroups = securityGroups
```

Comment:
in a function I usually don't assign values to member variables repetitively, because member vars are used to keep values across methods. Variables defined in this method are good enough to be written repetitively i think. But this is just my point of view.